tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -266,15 +278,18 @@ public function testOrderForRenderingWidgetAndFetchingSecureFields()
      */
     public function testRenderingWidgetWithEmptyName()
     {
-        $this->assertEquals([], $this->Form->fields);
+        $this->View->setRequest($this->View->getRequest()->withAttribute('formToken', []));
+        $this->Form->create();
 
         $result = $this->Form->widget('select', ['secure' => true, 'name' => '']);
         $this->assertSame('<select name=""></select>', $result);
-        $this->assertEquals([], $this->Form->fields);
+        $result = $this->getProtectedProperty($this->Form->getFormProtector(), 'fields');
+        $this->assertEquals([], $result);
 
         $result = $this->Form->widget('select', ['secure' => true, 'name' => '0']);
         $this->assertSame('<select name="0"></select>', $result);
-        $this->assertEquals(['0'], $this->Form->fields);
+        $result = $this->getProtectedProperty($this->Form->getFormProtector(), 'fields');
```

Comment:
Should FormProtector provide get methods for inspecting its state? That would save us using reflection.

Reply:
Except for tests I don't see a need to have getters for the properties. Perhaps just adding `__debugInfo()` would suffice?

Reply:
Yeah, pulling out of `__debugInfo()` is reasonable for tests to me.